### PR TITLE
update: updated COSE extended attributes logic

### LINKED ISSUE #88: bug: fail to verify COSE signature with extra attributes in the protected header
When the COSE signature contains extra attribute keys, which are integers not strings, the code reports error:
```
extendedAttributes key requires string type
```
Those keys should be ignored if not marked as critical, or report error if marked as critical. If they are known attributes, we may convert keys into strings so that it can fill in the `signature.Attribute` structure.

Related code:
https://github.com/notaryproject/notation-core-go/blob/b5df54c7465b0c1b177233108355b4e7314f1a14/signature/cose/envelope.go#L531-L544

### PR DESCRIPTION
**Pull request by patrickzheng200**
Fixed issue #88.

After this update:
1. JWS extended attribute keys still enforce string type.
2. COSE extended attribute keys will accept string and int type.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

**Comment by JeyJeyGao** [comment]
LGTM

**Comment by priteshbandi** [comment]
For visibility, added some feedback in https://github.com/notaryproject/notation-core-go/issues/88#issuecomment-1281639077

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by yizha1** [comment]
@rgnote @gokarnm any further comments on this PR. It has been 9 days now. Can we merge it with two approvals from one org? /cc @dtzar

**Comment by rgnote** [comment]
@yizha1 Added a comment on https://github.com/notaryproject/notation-core-go/issues/88

File: signature/jws/jws.go [hunk 1]
```diff
@@ -189,9 +189,16 @@ func getSignedAttributes(req *signature.SignRequest, algorithm string) (map[stri
...
 	// write extended signed attributes to the extAttrs map
 	for _, elm := range req.ExtendedSignedAttributes {
-		extAttrs[elm.Key] = elm.Value
+		key, ok := elm.Key.(string)
+		if !ok {
+			return nil, &signature.InvalidSignRequestError{Msg: "JWS envelope format only supports key of type string"}
+		}
+		if _, ok := extAttrs[key]; ok {
+			return nil, &signature.InvalidSignRequestError{Msg: fmt.Sprintf("%q already exists in the extAttrs", key)}
+		}
```
Comment (shizhMSFT) [inline signature/jws/jws.go:198]: nice catch!

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Pull request merged**